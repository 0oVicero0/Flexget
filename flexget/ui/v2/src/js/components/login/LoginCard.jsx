import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Button from 'material-ui/Button';
import Card, { CardActions, CardContent } from 'material-ui/Card';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import TextField from 'components/common/TextField';

const styleSheet = createStyleSheet('LoginCard', theme => ({
  card: {
    maxWidth: 400,
    margin: '0 10px',
    [theme.breakpoints.up('sm')]: {
      margin: '0 auto',
    },
  },
  error: {
    color: theme.palette.error[500],
    textAlign: 'center',
    padding: 10,
  },
  button: {
    width: '100%',
  },
}));

const LoginCard = ({ classes, handleSubmit, errorStatus }) => (
  <Card className={classes.card}>
    <form onSubmit={handleSubmit}>
      <CardContent>
        <div className={classes.error}>
          {errorStatus.message}
        </div>
        <Field
          name="username"
          component={TextField}
          id="username"
          label="Username"
        />
        <Field
          name="password"
          component={TextField}
          id="password"
          label="Password"
          type="Password"
        />
      </CardContent>
      <CardActions>
        <Button type="submit" className={classes.button}>
          Login
        </Button>
      </CardActions>
    </form>
  </Card>
);

LoginCard.propTypes = {
  classes: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  errorStatus: PropTypes.object,
};

LoginCard.defaultProps = {
  errorStatus: {},
};

export default reduxForm({
  form: 'login',
})(withStyles(styleSheet)(LoginCard));
