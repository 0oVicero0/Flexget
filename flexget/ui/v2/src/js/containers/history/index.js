import { connect } from 'react-redux';
import HistoryPage from 'components/history';
import { request } from 'utils/actions';
import { GET_HISTORY } from 'store/history/actions';

function mapDispatchToProps(dispatch) {
  return {
    getHistory: data => dispatch(request(GET_HISTORY, data)),
  };
}

export default connect(null, mapDispatchToProps)(HistoryPage);
