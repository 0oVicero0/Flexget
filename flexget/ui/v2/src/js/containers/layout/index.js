import { connect } from 'react-redux';
import { checkLogin } from 'actions/auth';
import Layout from 'components/layout';

export function mapStateToProps({ auth }) {
  return {
    loggedIn: !!auth.loggedIn,
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    checkLogin: () => dispatch(checkLogin()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Layout);
