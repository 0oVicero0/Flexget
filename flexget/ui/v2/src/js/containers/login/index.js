import { connect } from 'react-redux';
import LoginPage from 'components/login';
import { GET_VERSION } from 'actions/version';

export function mapStateToProps({ auth, status }) {
  return {
    redirectToReferrer: !!auth.loggedIn,
    loading: !!status.loading[GET_VERSION],
  };
}

export default connect(mapStateToProps)(LoginPage);
