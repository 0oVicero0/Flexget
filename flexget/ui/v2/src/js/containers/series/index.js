import { connect } from 'react-redux';
import {
  GET_SHOWS,
} from 'actions/series/shows';
import SeriesPage from 'components/series';
import { request } from 'utils/actions';

function mapStateToProps({ series }) {
  return {
    shows: series.shows.items,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getShows: payload => dispatch(request(GET_SHOWS, payload)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(SeriesPage);
