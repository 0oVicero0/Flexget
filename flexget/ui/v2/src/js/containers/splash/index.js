import { connect } from 'react-redux';
import Splash from 'components/splash';
import { LOGIN, checkLogin } from 'actions/auth';

export function mapStateToProps({ status }) {
  return {
    checking: !!status.loading[LOGIN],
  };
}

export function mapDispatchToProps(dispatch) {
  return {
    checkLogin: () => dispatch(checkLogin()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Splash);
