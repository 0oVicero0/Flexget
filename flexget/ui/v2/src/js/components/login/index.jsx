import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import headerImage from 'images/header.png';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import LoginCard from 'containers/login/LoginCard';

const styleSheet = createStyleSheet('LoginPage', theme => ({
  logo: {
    background: `transparent url(${headerImage}) no-repeat center`,
    minHeight: 90,
    backgroundSize: '100% auto',
    margin: '0 10px',
    [theme.breakpoints.up('sm')]: {
      backgroundSize: 'auto',
    },
  },
}));

const LoginPage = ({ classes, redirectToReferrer, location }) => {
  const { from } = location.state || { from: { pathname: '/' } };

  if (redirectToReferrer) {
    return (
      <Redirect to={from} />
    );
  }

  return (
    <div>
      <div className={classes.logo} />
      <LoginCard />
    </div>
  );
};

LoginPage.propTypes = {
  classes: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  redirectToReferrer: PropTypes.bool.isRequired,
};

export default withStyles(styleSheet)(LoginPage);
