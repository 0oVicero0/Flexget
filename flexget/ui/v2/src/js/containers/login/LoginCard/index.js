import { connect } from 'react-redux';
import { LOGIN } from 'actions/auth';
import { request } from 'utils/actions';
import LoginCard from 'components/login/LoginCard';

export function mapStateToProps({ status }) {
  return {
    initialValues: {
      username: 'flexget',
    },
    errorStatus: status.error || {},
  };
}

function mapDispatchToProps(dispatch) {
  return {
    onSubmit: credentials => dispatch(request(LOGIN, credentials)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginCard);
