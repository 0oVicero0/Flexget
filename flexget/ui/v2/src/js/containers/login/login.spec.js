import { mapStateToProps } from 'containers/login';

describe('containers/login', () => {
  it('should return logged in if logged in', () => {
    expect(mapStateToProps({
      auth: { loggedIn: true },
      status: { loading: {} },
    })).toMatchSnapshot();
  });

  it('should return not logged in if logged out', () => {
    expect(mapStateToProps({ auth: { }, status: { loading: {} } })).toMatchSnapshot();
  });
});
