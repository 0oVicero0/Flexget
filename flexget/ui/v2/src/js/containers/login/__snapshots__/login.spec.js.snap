// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/login should return logged in if logged in 1`] = `
Object {
  "loading": false,
  "redirectToReferrer": true,
}
`;

exports[`containers/login should return not logged in if logged out 1`] = `
Object {
  "loading": false,
  "redirectToReferrer": false,
}
`;
